### PR TITLE
allow specifying engine version

### PR DESCRIPTION
**Pull request by james-lawrence**
also improves logging a bit for new users.

main thing is it uses the engine version, not the flutter version... perhaps it should use the flutter version may be easier to use?

### CODE REVIEW

File: internal/enginecache/cache.go [hunk 1]
```diff
@@ -214,11 +214,13 @@ func ValidateOrUpdateEngineAtPath(targetOS string, cachePath string) (engineCach
...
 	cachedEngineVersion := string(cachedEngineVersionBytes)
-	requiredEngineVersion := FlutterRequiredEngineVersion()
+	if len(requiredEngineVersion) == 0 {
+		requiredEngineVersion = FlutterRequiredEngineVersion()
+	}
 
 	if cachedEngineVersion != "" {
 		if cachedEngineVersion == requiredEngineVersion {
-			log.Printf("Using engine from cache")
+			log.Printf("Using engine from cache: %s/%s", cachedEngineVersionPath, cachedEngineVersion)
```
Comment (jld3103) [inline internal/enginecache/cache.go]: Maybe this is to verbose? The normal user doesn't need to know the used engine version and a user who overwrote the engine version does know which engine version they want to use. I'd keep it like before.

Comment (jld3103) [reply internal/enginecache/cache.go]: otherwise LGTM

**Comment by jld3103** [comment]
Maybe you could also add an option to the config file? Look at https://github.com/go-flutter-desktop/hover/blob/a6f11fe3c4946c9e7cce75360902bd79a077ccf4/internal/config/config.go#L26 and https://github.com/go-flutter-desktop/hover/blob/a6f11fe3c4946c9e7cce75360902bd79a077ccf4/cmd/build.go#L341

File: internal/enginecache/cache.go (hunk 1 above)
Comment (james-lawrence) [reply internal/enginecache/cache.go]: you okay with putting as a debug log? i had to go digging to find this stuff =)

Comment (jld3103) [reply internal/enginecache/cache.go]: Yep

File: internal/config/config.go [hunk 2]
```diff
@@ -29,6 +32,7 @@ type Config struct {
 	Branch    string
 	CachePath string `yaml:"cache-path"`
 	OpenGL    string
+	Engine    string
```
Comment (jld3103) [inline internal/config/config.go]: Would change it to `engine-version`

File: cmd/run.go [hunk 3]
```diff
@@ -3,14 +3,15 @@ package cmd
...
 	"fmt"
-	"github.com/go-flutter-desktop/hover/internal/config"
 	"io"
 	"os"
 	"os/exec"
 	"path/filepath"
 	"regexp"
 	"runtime"
 
+	"github.com/go-flutter-desktop/hover/internal/config"
```
Comment (jld3103) [inline cmd/run.go]: Sort imports

File: cmd/build.go [hunk 4]
```diff
@@ -385,9 +391,14 @@ func buildNormal(targetOS string, vmArguments []string) {
...
 
-	var trackWidgetCreation string
+	var (
+		trackWidgetCreation string
+		verbose             string
+	)
+
 	if buildDebug {
 		trackWidgetCreation = "--track-widget-creation"
+		verbose = "-v"
```
Comment (jld3103) [inline cmd/build.go]: This is to verbose again. If you want to enable it you need to add a flag to hover. But that Should be in another PR, but actually could be used in a lot of cases.

File: cmd/build.go [hunk 5]
```diff
@@ -348,6 +350,9 @@ func buildNormal(targetOS string, vmArguments []string) {
 	if buildCachePath == config.BuildCachePathDefault && config.GetConfig().CachePath != "" {
 		buildCachePath = config.GetConfig().CachePath
 	}
+	if buildEngineVersion == config.BuildEngineDefault && config.GetConfig().Engine != "" {
+		buildEngineVersion = config.GetConfig().Engine
+	}
```
Comment (jld3103) [inline cmd/build.go]: Add a warning that there could be errors if the engine version doesn't match with the one used by flutter.

File: cmd/build.go (hunk 4 above)
Comment (james-lawrence) [reply cmd/build.go]: I was thinking about that. I didn't actually mean for it to be committed in this pr.

File: internal/config/config.go (hunk 2 above)
Comment (jld3103) [reply internal/config/config.go]: Could you also update the template `hover.yaml`

Comment (james-lawrence) [reply internal/config/config.go]: done.

**Comment by jld3103** [comment]
LGTM
